### PR TITLE
Add package: ctodo

### PR DESCRIPTION
**Pull request by matthiasbeyer**
`ctodo` is a curses commandline todolist manager.

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4540,6 +4540,11 @@ let
     inherit pkgs ctags writeScriptBin;
   };
 
+  ctodo = callPackage ../applications/misc/ctodo {
+    cmake = cmake;
+    ncurses = ncurses;
```
Comment (vbgl) [inline pkgs/top-level/all-packages.nix]: Why are these two lines `x = x` needed? `callPackage` does this already.

File: pkgs/applications/misc/ctodo/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, cmake, fetchgit, ncurses }:
+
+let
+  version = "1.1";
+in
+stdenv.mkDerivation {
+  name = "ctodo-${version}";
+
+  src = fetchgit {
+    url = "https://github.com/Acolarh/ctodo.git";
```
Comment (iElectric) [inline pkgs/applications/misc/ctodo/default.nix]: Please use a tarball over git: https://github.com/Acolarh/ctodo/archive/v1.1.tar.gz

File: pkgs/applications/misc/ctodo/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  configurePhase = ''
+    cmake -DCMAKE_INSTALL_PREFIX=$out .
+  '';
+
+  meta = {
+    homepage = "http://ctodo.apakoh.dk/";
+    description = "A simple ncurses-based task list manager.";
+    license = stdenv.lib.licenses.mit;
+    maintainers = [ stdenv.lib.maintainers.matthiasbeyer ];
```
Comment (iElectric) [inline pkgs/applications/misc/ctodo/default.nix]: missing `meta.platforms`

**Comment by lethalman** [comment]
When you commit to the PR please comment, we don't get notified otherwise. Travis passed, thanks.

**Pull request merged**